### PR TITLE
Fix new lines in LSP server diagnostics.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/lsp/protocol_compiler_test_slow.toit [hunk 1]
```diff
@@ -245,3 +245,20 @@ test client/LspClient:
   expect (response.contains "code")
   UNKNOWN_ERROR_CODE ::= -32601
   expect_equals UNKNOWN_ERROR_CODE response["code"]
+
+  // Check that diagnostics have new lines.
```
Comment (kasperl) [inline tests/lsp/protocol_compiler_test_slow.toit:249]:
```suggestion
  // Check that diagnostics have newlines.
```

File: tests/lsp/protocol_compiler_test_slow.toit [hunk 2]
```diff
@@ -245,3 +245,20 @@ test client/LspClient:
...
+  print "Checking diagnostics have new lines"
```
Comment (kasperl) [inline tests/lsp/protocol_compiler_test_slow.toit:250]:
```suggestion
  print "Checking diagnostics have newlines"
```

File: tests/lsp/protocol_compiler_test_slow.toit [hunk 3]
```diff
@@ -245,3 +245,20 @@ test client/LspClient:
...
+  new_line_path := "$(directory.cwd)/new_line.toit"
```
Comment (kasperl) [inline tests/lsp/protocol_compiler_test_slow.toit:251]:
```suggestion
  newline_path := "$(directory.cwd)/newline.toit"
```

File: tests/lsp/protocol_compiler_test_slow.toit [hunk 4]
```diff
@@ -245,3 +245,20 @@ test client/LspClient:
...
+
+  // The diagnostic here will have multiple lines, as it
+  // explains how the 'foo' method could be called.
+  client.send_did_open --path=new_line_path --text="""
```
Comment (kasperl) [inline tests/lsp/protocol_compiler_test_slow.toit:255]:
```suggestion
  client.send_did_open --path=newline_path --text="""
```

File: tests/lsp/protocol_compiler_test_slow.toit [hunk 5]
```diff
@@ -245,3 +245,20 @@ test client/LspClient:
...
+    foo x y:
+    main:
+      foo 1
+    """
+  diagnostics = client.diagnostics_for --path=new_line_path
```
Comment (kasperl) [inline tests/lsp/protocol_compiler_test_slow.toit:260]:
```suggestion
  diagnostics = client.diagnostics_for --path=newline_path
```

File: tools/lsp/server/compiler.toit [hunk 6]
```diff
@@ -198,7 +198,8 @@ class Compiler:
             while true:
               line = reader.read_line
               if line == "*******************": break
-              msg += line
+              if msg != "": msg += "\n"
+              msg += "$line"
```
Comment (kasperl) [inline tools/lsp/server/compiler.toit:202]: I'd use a list of lines and join them at the end.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/lsp/server/compiler.toit (hunk 6 above)
Comment (floitsch) [reply tools/lsp/server/compiler.toit]: done.

**Pull request merged**